From f5f452c8314e864d3f8a4feeda76abb136a341d9 Mon Sep 17 00:00:00 2001
From: "sungmok.shin" <sungmok.shin@lge.com>
Date: Wed, 15 Jul 2020 15:21:08 +0900
Subject: [PATCH] Flush processor instruction cash

:Release Notes:
Flush processor instruction cash to shared memory.

:Detailed Notes:
Regarding
http://community.arm.com/groups/processors/blog/2010/02/17/caches-and-self-modifying-code
the best way creating self modifiable code in linux is use library
function __clear_cash.

:Testing Performed:
Add syslog in modified section of code and exec system services
with apropriate functionality.

:QA Notes:

:Issues Addressed:
[PLAT-493] Fix Patch on Flush function for CPU ARM

Change-Id: Ifb819ef4ac7d16366e81bb4b0f3ca5eb8f0d3bb9
---
 deps/v8/src/codegen/arm/cpu-arm.cc | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/deps/v8/src/codegen/arm/cpu-arm.cc b/deps/v8/src/codegen/arm/cpu-arm.cc
index b07c842b..fe402365 100644
--- a/deps/v8/src/codegen/arm/cpu-arm.cc
+++ b/deps/v8/src/codegen/arm/cpu-arm.cc
@@ -29,6 +29,8 @@ V8_NOINLINE void CpuFeatures::FlushICache(void* start, size_t size) {
 #if !defined(USE_SIMULATOR)
 #if V8_OS_QNX
   msync(start, size, MS_SYNC | MS_INVALIDATE_ICACHE);
+#elif defined(__GNUC__)
+  __builtin___clear_cache(start, start+size);
 #else
   const int N_RETRIES = 100;
   int err = 0;
-- 
2.17.1

